include/seastar/core/bitset-iter.hh

```diff
@@ -32,7 +35,7 @@ static constexpr int ulong_bits = std::numeric_limits<unsigned long>::digits;
  * which is returned when value == 1.
  */
 template<typename T>
-inline size_t count_leading_zeros(T value) noexcept;
+size_t count_leading_zeros(T value) noexcept;
```

Comment:
This smells like a compiler bug, no? I don't mind the change, but if it's a bug, let's report it.

Reply:
yeah, i just was juist not 100% sure if it was indeed a bug. as C++ modules is quite a new thing to me. wanted to collect more facts or at least do my home work before filing an issue. anyhow, here it is: https://github.com/llvm/llvm-project/issues/62043

Reply:
will extract it into a separate commit.

Reply:
i put it back in the latest revision. as this behavior is by design. i amended the commit message to explain the rationales behind the change.

Reply:
ok, so the cause is that is SEASTAR_MODULE_EXPORT_BEGIN doesn't export anything, then count_trailing_zeros us undefined? But then how does the module see it?

Reply:
> ok, so the cause is that is SEASTAR_MODULE_EXPORT_BEGIN doesn't export anything,

not quite true. in our case, `SEASTAR_MODULE_EXPORT_BEGIN` is defined as `export {`. so all stuff after it before ``SEASTAR_MODULE_EXPORT_END` is exported.

> then count_trailing_zeros us undefined? But then how does the module see it?

good question. i gave up too soon. let me continue the discussion with Chuanqi at https://github.com/llvm/llvm-project/issues/62043

Reply:
okay. i mistook the specialization of a template as its definition. please refer to the link above for more details.